storage/innobase/buf/buf0flu.cc

```diff
@@ -2220,6 +2258,9 @@ static os_thread_ret_t DECLARE_THREAD(buf_flush_page_cleaner)(void*)
 next:
 #endif /* !DBUG_OFF */
     mysql_mutex_lock(&buf_pool.flush_list_mutex);
+
+    if (idle_flush)
+      buf_pool.page_cleaner_set_idle(true);
```

Comment:
Is this change really needed? If yes, why? I would expect `goto unemployed` to do the right thing. Note: during the idle flushing, we may write multiple batches of `srv_io_capacity` (`innodb_io_capacity`) pages in this loop.

Reply:
Thank you. I now realized that if we fail to call this, then at the start of the next loop we would invoke `my_cond_timedwait()` and cause an unnecessary 1-second wakeup.